### PR TITLE
boards*: remove remaining uses of export with some openocd related variables

### PR DESCRIPTION
**Pull request by aabadie**
### Contribution description

This PR removes remaining use of global `export` for openocd related variables. It is similar to #13452 and #13480.

Build system static check is also adapted.

### Testing procedure

- A green CI/Murdock
- Verify that the adapted boards (nrf52 when softdevice is used, phynode-kw41z, mulle, seeduino_arch-pro, pba-xxx) can still be flashed with openocd
- Verify reset still work on hifive1  (who has that boards ?)

### Issues/PRs references

Related to #10850

### CODE REVIEW

**Comment by kaspar030** [comment]
I don't have any of the affected boards, so I'm trying to verify if openocd.sh is still getting the parameters, by manually checking its output with `set -x` added at the top.

I'll add what I tested as comments to the diff.

File: boards/common/frdm/Makefile.include [hunk 1]
```diff
@@ -14,7 +14,7 @@ DEBUG_ADAPTER ?= dap
 OPENOCD_CONFIG ?= $(RIOTBOARD)/common/frdm/dist/openocd-$(CPU_FAMILY).cfg
 
 # Check the flash configuration field before flashing
-export PRE_FLASH_CHECK_SCRIPT = $(RIOTCPU)/$(CPU)/dist/check-fcfield.sh
+PRE_FLASH_CHECK_SCRIPT = $(RIOTCPU)/$(CPU)/dist/check-fcfield.sh
```
Comment (kaspar030) [inline boards/common/frdm/Makefile.include:17]:
```
[kaspar@ng riot.tmp (pr13719)]$ BOARD=frdm-kw41z make -Cexamples/hello-world flash-only 2>&1 | grep check-fcfield
+ '[' -n /home/kaspar/src/riot.tmp/cpu/kinetis/dist/check-fcfield.sh ']'
+ sh -c '/home/kaspar/src/riot.tmp/cpu/kinetis/dist/check-fcfield.sh '\''/home/kaspar/src/riot.tmp/examples/hello-world/bin/frdm-kw41z/hello-world.elf'\'''
[kaspar@ng riot.tmp (pr13719)]$ 
```

:+1:

File: boards/common/nrf52/Makefile.include [hunk 2]
```diff
@@ -40,8 +40,8 @@ else ifeq (openocd,$(PROGRAMMER))
   ifneq (,$(filter nordic_softdevice_ble,$(USEPKG)))
     LINKER_SCRIPT ?= $(RIOTCPU)/$(CPU)/ldscripts/$(CPU_MODEL)_sd.ld
 
-    export OPENOCD_PRE_FLASH_CMDS += -c 'flash write_image erase "$(BINDIR)/softdevice.hex"'
-    export OPENOCD_PRE_VERIFY_CMDS += -c 'verify_image "$(BINDIR)/softdevice.hex"'
+    OPENOCD_PRE_FLASH_CMDS += -c 'flash write_image erase "$(BINDIR)/softdevice.hex"'
```
Comment (kaspar030) [inline boards/common/nrf52/Makefile.include:43]: still there :+1:

File: boards/common/nrf52/Makefile.include [hunk 3]
```diff
@@ -40,8 +40,8 @@ else ifeq (openocd,$(PROGRAMMER))
...
+    OPENOCD_PRE_VERIFY_CMDS += -c 'verify_image "$(BINDIR)/softdevice.hex"'
```
Comment (kaspar030) [inline boards/common/nrf52/Makefile.include:44]: also there

File: boards/hifive1/Makefile.include [hunk 4]
```diff
@@ -6,7 +6,7 @@ PORT_DARWIN ?= $(firstword $(sort $(wildcard /dev/tty.usbmodem*)))
 include $(RIOTMAKE)/tools/serial.inc.mk
 
 # this board uses openocd with a custom reset command
-export OPENOCD_CMD_RESET_RUN=-c _reset
+OPENOCD_CMD_RESET_RUN =-c _reset
```
Comment (kaspar030) [inline boards/hifive1/Makefile.include:9]: :+1:

File: boards/mulle/Makefile.include [hunk 5]
```diff
@@ -17,10 +17,10 @@ PORT_DARWIN ?= $(firstword $(sort $(wildcard /dev/tty.usbserial*)))
 # verification by using the MCU to compute the image checksum after flashing.
 # wdog-disable.bin is a precompiled binary which will disable the watchdog and
 # return control to the debugger (OpenOCD)
-export OPENOCD_PRE_VERIFY_CMDS += \
+OPENOCD_PRE_VERIFY_CMDS += \
   -c 'load_image $(RIOTCPU)/$(CPU)/dist/wdog-disable.bin 0x20000000 bin' \
   -c 'resume 0x20000000'
-export PRE_FLASH_CHECK_SCRIPT = $(RIOTCPU)/$(CPU)/dist/check-fcfield.sh
+PRE_FLASH_CHECK_SCRIPT = $(RIOTCPU)/$(CPU)/dist/check-fcfield.sh
```
Comment (kaspar030) [inline boards/mulle/Makefile.include:23]: not set anymore

File: boards/mulle/Makefile.include [hunk 6]
```diff
@@ -17,10 +17,10 @@ PORT_DARWIN ?= $(firstword $(sort $(wildcard /dev/tty.usbserial*)))
 # verification by using the MCU to compute the image checksum after flashing.
 # wdog-disable.bin is a precompiled binary which will disable the watchdog and
 # return control to the debugger (OpenOCD)
-export OPENOCD_PRE_VERIFY_CMDS += \
+OPENOCD_PRE_VERIFY_CMDS += \
```
Comment (kaspar030) [inline boards/mulle/Makefile.include:20]: not set anymore

File: boards/pba-d-01-kw2x/Makefile.include [hunk 7]
```diff
@@ -6,10 +6,10 @@ PORT_DARWIN ?= $(firstword $(sort $(wildcard /dev/tty.usbmodem*)))
 # verification by using the MCU to compute the image checksum after flashing.
 # wdog-disable.bin is a precompiled binary which will disable the watchdog and
 # return control to the debugger (OpenOCD)
-export OPENOCD_PRE_VERIFY_CMDS += \
+OPENOCD_PRE_VERIFY_CMDS += \
   -c 'load_image $(RIOTCPU)/$(CPU)/dist/wdog-disable.bin 0x20000000 bin' \
   -c 'resume 0x20000000'
-export PRE_FLASH_CHECK_SCRIPT = $(RIOTCPU)/$(CPU)/dist/check-fcfield.sh
+PRE_FLASH_CHECK_SCRIPT = $(RIOTCPU)/$(CPU)/dist/check-fcfield.sh
```
Comment (kaspar030) [inline boards/pba-d-01-kw2x/Makefile.include:12]: these two are missing too

File: boards/phynode-kw41z/Makefile.include [hunk 8]
```diff
@@ -7,7 +7,7 @@ ifeq (openocd,$(PROGRAMMER))
 endif
 
 # Enable direct write to FCF (required for setting FOPT byte).
-export OPENOCD_PRE_FLASH_CMDS += "-c kinetis fcf_source write"
+OPENOCD_PRE_FLASH_CMDS += "-c kinetis fcf_source write"
```
Comment (kaspar030) [inline boards/phynode-kw41z/Makefile.include:10]: :+1:

File: boards/seeeduino_arch-pro/Makefile.include [hunk 9]
```diff
@@ -12,7 +12,7 @@ FLASHFILE ?= $(HEXFILE)
 include $(RIOTMAKE)/tools/openocd.inc.mk
 
 # generate image checksum from hex file
-export PRE_FLASH_CHECK_SCRIPT = lpc_checksum --format hex
+PRE_FLASH_CHECK_SCRIPT = lpc_checksum --format hex
 
 # remap flash to address 0x0, otherwise it verifies the rom bootloader
-export OPENOCD_PRE_VERIFY_CMDS += '-c reset init'
+OPENOCD_PRE_VERIFY_CMDS += '-c reset init'
```
Comment (kaspar030) [inline boards/seeeduino_arch-pro/Makefile.include:18]: both :+1:

**Comment by aabadie** [comment]
Thanks for spotting these problems @kaspar030. It turned out that there were some order issues with variable assignment and openocd makefile include for mulle, pba-xxx and seeeduino. They should be fixed now.

**Comment by kaspar030** [comment]
>  It turned out that there were some order issues with variable assignment and openocd makefile include for mulle, pba-xxx and seeeduino.

ouch, that sounds like it will happen again ;)

will re-check.

**Comment by aabadie** [comment]
> that sounds like it will happen again ;)

Indeed, it might even be present elsewhere already.

File: boards/mulle/Makefile.include (hunk 6 above)
Comment (kaspar030) [reply boards/mulle/Makefile.include:23]: :+1:

File: boards/mulle/Makefile.include (hunk 5 above)
Comment (kaspar030) [reply boards/mulle/Makefile.include:26]: :+1:

**Comment by kaspar030** [comment]
Ok, the missing ones are here now.

**Pull request merged**